engine/src/main.rs

```diff
@@ -21,20 +22,37 @@ fn main() {
         if let GameResult::Winner(hw) = history.result {
             if let GameResult::Winner(sw) = state.game_result {
                 if sw != hw {
-                    println!("winners don't match");
+                    return Err(GameError::ResultMismatch {
+                        reported_result: history.result,
+                        actual_result: state.game_result,
+                    });
                 }
             }
             if let GameResult::Draw = state.game_result {
-                println!("winners don't match");
+                return Err(GameError::ResultMismatch {
+                    reported_result: history.result,
+                    actual_result: state.game_result,
+                });
             }
         }
         if let GameResult::Draw = history.result {
             println!("History says game ended in a draw");
             if let GameResult::Winner(_) = state.game_result {
-                panic!("State says there is a winner");
+                return Err(GameError::ResultMismatch {
+                    reported_result: history.result,
+                    actual_result: state.game_result,
+                });
             }
         }
+        Ok(())
     } else {
-        println!("Please supply a .pgn to evaluate.")
+        Err(GameError::NoPgnFile)
+    }
+}
+
+fn main() {
+    match play_game() {
+        Ok(_) => {},
+        Err(e) => eprintln!("{}", e),
```

Comment:
for this, i'd just change `main`'s type signature to return a `Result<(), GameError>` and do `play_game()?`

Reply:
resolving this per discord convo!